source/collections/VectorList.ooc

```diff
@@ -123,11 +123,10 @@ VectorList: class <T> {
 		for (i in 0 .. this count)
 			this[i] = function(this[i])
 	}
-	map: func <T, S> (function: Func(T) -> S) -> This<S> {
+	map: func <S> (function: Func(T) -> S) -> This<S> {
 		result := This<S> new(this count)
-		for (i in 0 .. this count) {
+		for (i in 0 .. this count)
 			result add(function(this[i]))
-		}
 		result
 	}
 	fold: func <T, S> (function: Func(T, S) -> S, initial: S) -> S {
```

Comment:
If that *works*, it's better than before, which didn't generally work at all, even if you specified `S`. Is that the case now?